pkg/container/docker_run.go

```diff
@@ -568,25 +592,32 @@ func (cr *containerReference) exec(cmd []string, env map[string]string, user, wo
 			errWriter = os.Stderr
 		}
 
+		var err error
 		if !isTerminal || os.Getenv("NORAW") != "" {
 			_, err = stdcopy.StdCopy(outWriter, errWriter, resp.Reader)
 		} else {
 			_, err = io.Copy(outWriter, resp.Reader)
 		}
-		if err != nil {
-			logger.Error(err)
-		}
+		cmdResponse <- err
+	}()
 
-		inspectResp, err := cr.cli.ContainerExecInspect(ctx, idResp.ID)
+	select {
+	case <-ctx.Done():
+		// send ctrl + c
+		_, err := resp.Conn.Write([]byte{3})
```

Comment:
That could work as well if we also define an ENV-Var like `RUNNER_TRACKING_ID` and all processes inherits that, and as well we know exactly which containers we did start.

That is quite easy for `action/runners` but way harder for us, since users and have additional containers running which which we should not touch. On github-actions there are no other users processes.

Reply:
I'm not shure if you understood my comment..
You can ignore `RUNNER_TRACKING_ID` for this docker impl, this is only relevant for non container scenarios ( which act doesn't support ).

> On github-actions there are no other users processes.

Hmm, my linux server has all normal services side by side, the actions/runner don't touch them. Also I run multiple actions/runner on the same host.

Reply:
This change is not meant to just do the `Job Cleanup Step` of github.
I think it is important to handle a run cancelation in a user aware manner. What I'v meant above is that it is more complex to cleanup in a user system than on a controlled environment like `actions/runner`.